python/lsst/the/monster/isolate_and_transform.py

```diff
@@ -0,0 +1,149 @@
+    GaiaDR3CatInfoClass = GaiaDR3Info
+    testing_mode = False
+
+    def run(self,
+            htmid=None,
+            catalog_list=[GaiaXPInfo, SkyMapperInfo, PS1Info, VSTInfo],
+            write_path_inp=None,
+            gaia_cat_info_class=None
+            ):
+
+        # read in gaiaDR3 cat htmid
+        # Read in the Gaia stars in the htmid.
+        if gaia_cat_info_class is None:
+            gaia_info = self.GaiaDR3CatInfoClass()
+        else:
+            gaia_info = gaia_cat_info_class
+
+        gaia_stars_all = read_stars(gaia_info.path, [htmid], allow_missing=self.testing_mode)
+
+        # isolate the gaia cat
+        gaia_stars = self._remove_neighbors(
+            gaia_stars_all
+        )
+        # loop over other catalogs
+        for cat_info in catalog_list:
+            # catalog_list should be a list of
+            # cat_info = self.CatInfoClass() e.g. gaia cat
+
+            # output columns are target catalog id, gaia id, coordinates,
+            # and the des fluxes
+            outcols = ["id", gaia_info.name + "_id", "coord_ra", "coord_dec"]
+            outcols += [f"decam_{band}_flux_from_{cat_info().name}" for band in cat_info().bands]
+
+            # read in star cat (if it exists)
+            if os.path.isfile(cat_info().path+'/'+str(htmid)+'.fits'):
+                cat_stars = read_stars(cat_info().path, [htmid], allow_missing=self.testing_mode)
+
+                # match with gaia_stars
+                with Matcher(gaia_stars["coord_ra"], gaia_stars["coord_dec"]) as m:
+                    idx, i1, i2, d = m.query_knn(
+                        cat_stars["coord_ra"],
+                        cat_stars["coord_dec"],
+                        distance_upper_bound=0.5/3600.0,
+                        return_indices=True,
+                    )
+                cat_stars = cat_stars[i2]
+                cat_stars.add_column(gaia_stars["id"][i1], name=gaia_info.name + "_id")
+
+                for band in cat_info.bands:
+                    # yaml spline fits are per-band, so loop over bands
+                    # read in spline
+                    filename = os.path.abspath(os.path.dirname(__file__))[:-23]
+                    filename += 'colorterms/'+cat_info().name
+                    filename += '_to_DES_band_'+str(band)+'.yaml'
+
+                    colorterm_spline = ColortermSpline.load(filename)
+
+                    # apply colorterms to transform to des mag
+                    band_1, band_2 = cat_info().get_color_bands(band)
+                    model_flux = colorterm_spline.apply(
+                        cat_stars[cat_info().get_flux_field(band_1)],
+                        cat_stars[cat_info().get_flux_field(band_2)],
+                        cat_stars[cat_info().get_flux_field(band)],
+                    )
+
+                    # Append the modeled mags column to cat_stars
+                    cat_stars.add_column(model_flux, name=f"decam_{band}_flux_from_{cat_info().name}")
```

Comment:
So for refcat format the name should end with `flux`, so something like `f"decam_{band}_from_{cat_info.name}_flux"`.  And ALSO we need the errors: `f"decam_{band}_from_{cat_info.name}_fluxErr"`

To compute the errors, you use the input error (which I think will be `cat_info.get_flux_field(band) + "Err"` but now I'm not entirely sure ... we may need to have some special casing here depending on how the original sharding was done).  And then do something like `new_error = new_flux * (original_error / original_flux)` to maintain S/N because this is just a shift in the calibration.